File: medplum-typescript/packages/sdk/src/model/fhir/Timing_Repeat.ts
import { code } from "./code";
import { decimal } from "./decimal";
import { Duration } from "./Duration";
import { Extension } from "./Extension";
import { Period } from "./Period";
import { positiveInt } from "./positiveInt";
import { Range } from "./Range";
import { time } from "./time";
import { Timing_repeatDurationunit } from "./Timing_repeatDurationunit";
import { Timing_repeatPeriodunit } from "./Timing_repeatPeriodunit";
import { Timing_repeatWhenItem } from "./Timing_repeatWhenItem";
import { unsignedInt } from "./unsignedInt";

export interface Timing_Repeat {
    id: string | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    boundsDuration: Duration | null | undefined;
    boundsRange: Range | null | undefined;
    boundsPeriod: Period | null | undefined;
    count: positiveInt | null | undefined;
    countMax: positiveInt | null | undefined;
    duration: decimal | null | undefined;
    durationMax: decimal | null | undefined;
    durationUnit: Timing_repeatDurationunit | null | undefined;
    frequency: positiveInt | null | undefined;
    frequencyMax: positiveInt | null | undefined;
    period: decimal | null | undefined;
    periodMax: decimal | null | undefined;
    periodUnit: Timing_repeatPeriodunit | null | undefined;
    dayOfWeek: code[] | null | undefined;
    timeOfDay: time[] | null | undefined;
    when: Timing_repeatWhenItem[] | null | undefined;
    offset: unsignedInt | null | undefined;
}

File: medplum-java/medplum-java-model/src/main/java/com/fhir/Timing_Repeat.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableTiming_Repeat.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface Timing_Repeat {
  Optional<decimal> duration();

  Optional<String> id();

  Optional<positiveInt> frequency();

  Optional<Timing_repeatPeriodunit> periodUnit();

  Optional<positiveInt> count();

  Optional<List<code>> dayOfWeek();

  Optional<List<Extension>> modifierExtension();

  Optional<unsignedInt> offset();

  Optional<List<Timing_repeatWhenItem>> when();

  Optional<decimal> periodMax();

  Optional<Range> boundsRange();

  Optional<positiveInt> countMax();

  Optional<Duration> boundsDuration();

  Optional<Period> boundsPeriod();

  Optional<decimal> durationMax();

  Optional<Timing_repeatDurationunit> durationUnit();

  Optional<positiveInt> frequencyMax();

  Optional<List<Extension>> extension();

  Optional<decimal> period();

  Optional<List<time>> timeOfDay();

  static ImmutableTiming_Repeat.Builder builder() {
    return ImmutableTiming_Repeat.builder();
  }
}
